### PR TITLE
Add tests for SplitResponseProcessor

### PR DESCRIPTION
**Pull request by dbwiddis**
### Description

Adds a test exercising the SplitResponseProcessor, re-enabled in 2.17:
 - https://github.com/opensearch-project/OpenSearch/pull/15393
 - https://github.com/opensearch-project/OpenSearch/pull/15395

### Issues Resolved

Continuation of #440 superceded by #505

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dbwiddis** [comment]
Failing test pending merge and backport of PR linked above, but opening this PR is a prerequisite for merging that one. 🐔 🥚 
>  ERROR   RESPONSE STATUS (Expected status 200, but received 400: application/json. Invalid processor type split)

File: CHANGELOG.md [hunk 1]
```diff
@@ -71,14 +71,15 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
...
 - Added support for annotating and testing the API spec against multiple OpenSearch distributions ([#483](https://github.com/opensearch-project/opensearch-api-specification/pull/483))
 - Added `read_time`, `write_time`, `queue_size` and `io_time_in_millis` to `IoStatDevice` ([#483](https://github.com/opensearch-project/opensearch-api-specification/pull/483))
 - Added `total_rejections_breakup` to `ShardIndexingPressureStats` ([#483](https://github.com/opensearch-project/opensearch-api-specification/pull/483))
 - Added `cancelled_task_percentage` and `current_cancellation_eligible_tasks_count` to `ShardSearchBackpressureTaskCancellationStats` ([#483](https://github.com/opensearch-project/opensearch-api-specification/pull/483))
 - Added detailed test coverage report ([#513](https://github.com/opensearch-project/opensearch-api-specification/pull/513))
 - Added 404 responses to `/_alias/{name}` and `/{index}/_alias/{name}` ([#519](https://github.com/opensearch-project/opensearch-api-specification/pull/519))
+- Added tests for `SplitResponseProcessor` ([#526](https://github.com/opensearch-project/opensearch-api-specification/pull/526))
```
Comment (dblock) [inline CHANGELOG.md:82]: No need to mention tests.

**Review by dblock — COMMENTED** [review]
I merged the linked PR.

**Review by dblock — DISMISSED** [review]

**Comment by dbwiddis** [comment]
Still failing because workflow [spec files](https://github.com/opensearch-project/opensearch-api-specification/actions/runs/10555001957/workflow#L35-L41) don't match [current](https://hub.docker.com/r/opensearchstaging/opensearch/tags)

**Comment by dblock** [comment]
> Still failing because workflow [spec files](https://github.com/opensearch-project/opensearch-api-specification/actions/runs/10555001957/workflow#L35-L41) don't match [current](https://hub.docker.com/r/opensearchstaging/opensearch/tags)

You should update the sha in this PR.

**Pull request merged**